azurerm_cdn_frontdoor_rule url_redirect_action query_string must not be empty ERROR

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

<!--- Please keep this note for the community --->

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->


### Terraform Version

1.3.0

### AzureRM Provider Version

3.35.0

### Affected Resource(s)/Data Source(s)

azurerm_cdn_frontdoor_rule

### Terraform Configuration Files

```hcl
Defined the Rule with the following


#RULE SETS
  rule_sets = [
    {
      name                 = "RedirectHRCPToLogin"
      custom_resource_name = "custom_rule"
      rules = [
        {
          name                 = "redirect"
          custom_resource_name = "HRCPRedirect"
          order                = 1
          actions = {
            url_redirect_action = {
              redirect_type = "PermanentRedirect"
              destination_hostname = "XXXXXXXXX.azurewebsites.net"
              redirect_protol = "MatchRequest"
              destination_path   = "/HRCPWeb/jsp2/login2.jsp"              
            }
          }
          conditions = {
            request_uri_condition = {
              operator     = "Contains"
              match_values = ["hrcp"]
            }
          }
        }
      ]
    },
```

Resource is defined as follows for the redirect 
```
resource "azurerm_cdn_frontdoor_rule" "frontdoor_rule" {
  depends_on = [azurerm_cdn_frontdoor_origin_group.frontdoor_origin_group, azurerm_cdn_frontdoor_origin.frontdoor_origin]

  for_each = try({ for rule in local.rules_per_rule_set : format("%s.%s", rule.rule_set_name, rule.name) => rule }, {})

  name                      = each.value.name
  cdn_frontdoor_rule_set_id = azurerm_cdn_frontdoor_rule_set.frontdoor_rule_set[each.value.rule_set_name].id
  order                     = each.value.order
  behavior_on_match         = each.value.behavior_on_match

  actions {
    dynamic "url_rewrite_action" {
      for_each = each.value.actions.url_rewrite_action == null ? [] : ["enabled"]
      content {
        source_pattern          = try(each.value.actions.url_rewrite_action.source_pattern, null)
        destination             = try(each.value.actions.url_rewrite_action.destination, null)
        preserve_unmatched_path = try(each.value.actions.url_rewrite_action.preserve_unmatched_path, null)
      }
    }
    dynamic "url_redirect_action" {
      for_each = each.value.actions.url_redirect_action == null ? [] : ["enabled"]
      content {
        redirect_type        = try(each.value.actions.url_redirect_action.redirect_type, null)
        destination_hostname = try(each.value.actions.url_redirect_action.destination_hostname, null)
        redirect_protocol    = try(each.value.actions.url_redirect_action.redirect_protocol, null)
        destination_path     = try(each.value.actions.url_redirect_action.destination_path, null)
        query_string         = try(each.value.actions.url_redirect_action.query_string, null)
        destination_fragment = try(each.value.actions.url_redirect_action.destination_fragment, null)
      }
    }
.......
```
```


### Debug Output/Panic Output

```shell
╷
│ Error: 'url_redirect_action' is invalid: "actions.0.url_redirect_action.0.query_string" must not be empty, got ""
│ 
│   with module.frontdoor_standard[0].azurerm_cdn_frontdoor_rule.frontdoor_rule["RedirectHRCPToLogin.redirect"],
│   on modules/Frontdoor/rules.tf line 18, in resource "azurerm_cdn_frontdoor_rule" "frontdoor_rule":
│   18:   actions {
│ 
╵
##[error]Bash exited with code '1'.
```


### Expected Behaviour

**The rule should be created with an empty query_string per the documentation** 

[query_string](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/cdn_frontdoor_rule#query_string) - (Optional) The query string used in the redirect URL. The value must be in the <key>=<value> or <key>={action_server_variable} format and must not include the leading ?, leave blank to preserve the incoming query string. Maximum allowed length for this field is 2048 characters. Defaults to an empty string.

### Actual Behaviour

The resource fails with the error that query_string cannot be empty

```
╷
│ Error: 'url_redirect_action' is invalid: "actions.0.url_redirect_action.0.query_string" must not be empty, got ""
│ 
│   with module.frontdoor_standard[0].azurerm_cdn_frontdoor_rule.frontdoor_rule["RedirectHRCPToLogin.redirect"],
│   on modules/Frontdoor/rules.tf line 18, in resource "azurerm_cdn_frontdoor_rule" "frontdoor_rule":
│   18:   actions {
│ 
╵
##[error]Bash exited with code '1'.
```


### Steps to Reproduce

_No response_

### Important Factoids

_No response_

### References

_No response_

--- a/internal/services/cdn/cdn_frontdoor_rule_resource.go
+++ b/internal/services/cdn/cdn_frontdoor_rule_resource.go
@@ -123,7 +123,7 @@ func resourceCdnFrontDoorRule() *pluginsdk.Resource {
 									},
 
 									// NOTE: it is valid for the query string to be an empty string.
-									// Leave blank to preserve the incoming query string. Issue #18249
+									// Leave blank to preserve the incoming query string. Issue #18249 & #19682
 									"query_string": {
 										Type:     pluginsdk.TypeString,
 										Optional: true,

--- a/internal/services/cdn/cdn_frontdoor_rule_resource_test.go
+++ b/internal/services/cdn/cdn_frontdoor_rule_resource_test.go
@@ -330,6 +330,21 @@ func TestAccCdnFrontDoorRule_honorOrigin(t *testing.T) {
 	})
 }
 
+func TestAccCdnFrontDoorRule_allowEmptyQueryString(t *testing.T) {
+	// NOTE: Regression test case for issue #19682
+	data := acceptance.BuildTestData(t, "azurerm_cdn_frontdoor_rule", "test")
+	r := CdnFrontDoorRuleResource{}
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.allowEmptyQueryString(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
 func (r CdnFrontDoorRuleResource) Exists(ctx context.Context, clients *clients.Client, state *pluginsdk.InstanceState) (*bool, error) {
 	id, err := parse.FrontDoorRuleID(state.ID)
 	if err != nil {
@@ -1117,5 +1132,43 @@ resource "azurerm_cdn_frontdoor_rule" "test" {
     }
   }
 }
+`, template, data.RandomInteger)
+}
+
+func (r CdnFrontDoorRuleResource) allowEmptyQueryString(data acceptance.TestData) string {
+	template := r.template(data)
+	return fmt.Sprintf(`
+  provider "azurerm" {
+    features {}
+  }
+  
+  %s
+  
+  resource "azurerm_cdn_frontdoor_rule" "test" {
+    depends_on = [azurerm_cdn_frontdoor_origin_group.test, azurerm_cdn_frontdoor_origin.test]
+  
+    name                      = "accTestRule%d"
+    cdn_frontdoor_rule_set_id = azurerm_cdn_frontdoor_rule_set.test.id
+  
+    order = 0
+  
+    conditions {
+      request_uri_condition {
+        match_values     = ["contoso"]
+        negate_condition = false
+        operator         = "Contains"
+      }
+    }
+  
+    actions {
+      url_redirect_action {
+        redirect_type        = "PermanentRedirect"
+        redirect_protocol    = "MatchRequest"
+        query_string         = ""
+        destination_hostname = "contoso.com"
+        destination_path     = "/test/page"
+      }
+    }
+  }
 `, template, data.RandomInteger)
 }

--- a/internal/services/cdn/validate/front_door_validation_helpers.go
+++ b/internal/services/cdn/validate/front_door_validation_helpers.go
@@ -141,8 +141,6 @@ func CdnFrontDoorUrlRedirectActionQueryString(i interface{}, k string) (_ []stri
 		if len(v) > 2048 {
 			return nil, []error{fmt.Errorf("'url_redirect_action' is invalid: %q cannot be longer than 2048 characters in length, got %d", k, len(v))}
 		}
-	} else {
-		return nil, []error{fmt.Errorf("'url_redirect_action' is invalid: %q must not be empty, got %q", k, v)}
 	}
 
 	return nil, nil

